src/com/carolinarollergirls/scoreboard/core/ScoreBoard.java

```diff
@@ -106,6 +107,7 @@
     public Media getMedia();
 
     public enum Value implements PermanentProperty {
+        VERSION(String.class, Version.get()),
```

Comment:
Hmm, is there a race here? If this class is loaded before loadVersion runs it'll be empty. I'd suggest passing the version down as an argument when the ScoreBoard is created.

Reply:
No need to use an argument when the value is available through a static function.

Reply:
That basically the long way of having a global variable, pass it down would be cleaner I think.

Reply:
Well, the value is inherently global. If you want to pass it as parameter, you also have to pass it through multiple layers to the version servlet. And then do the same with the base path and the logger. A huge amount of extra code just to follow some dogma.

Reply:
The interplay between the ScoreBoardManager and Version smells a bit. If all the version parsing code was in the class that'd be okay, the split seems arbitrary currently.

Reply:
The scope creep continues, but this should now resolve any and all layering violations that involve ScoreBoardManager.
(I had to split off the Logger as the Version parsing code used it and after that there wasn't much left anyways.)

Reply:
Thanks. Logging needs a complete re-work, that's something for another day.